tuf/tuf.go

```diff
@@ -173,6 +173,132 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 	return nil
 }
 
+// GetBaseRole gets a base role from this repo's metadata
+func (tr *Repo) GetBaseRole(name string) (data.BaseRole, error) {
+	if !data.IsBaseRole(name) {
+		return data.BaseRole{}, fmt.Errorf("Invalid base role name specified: %s", name)
+	}
+	roleData, ok := tr.Root.Signed.Roles[name]
```

Comment:
Should we check to see if `tr.Root == `nil` first?  Otherwise this might panic.  Similarly for `GetDelegationRole`